### PR TITLE
Support explicitly setting default database

### PR DESCRIPTION
**Pull request by jnfeinstein**
This solves one aspect of #1125 by allowing the user to explicitly set a default database that is not affected by adding additional connections.

### CODE REVIEW

**Comment by jnfeinstein** [comment]
@Tapac I was unsure how to test this code.  It is not clear to me where to put the test nor how to make a connection to the local test db.  In addition, I think it is impossible to _reset_ the `TransactionManager` class by nature of companion objects.

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/transactions/TransactionApi.kt [hunk 1]
```diff
@@ -55,23 +56,27 @@ interface TransactionManager {
     fun bindTransactionToThread(transaction: Transaction?)
 
     companion object {
+        private val currentDefaultDatabase = AtomicReference<Database>()
 
-        private val managers = ConcurrentLinkedDeque<TransactionManager>().apply {
-            push(NotInitializedManager)
-        }
+        var defaultDatabase: Database?
+            get() { return currentDefaultDatabase.get() ?: databases.firstOrNull() }
```
Comment (Tapac) [inline exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/transactions/TransactionApi.kt:62]:
```suggestion
            get() = currentDefaultDatabase.get() ?: databases.firstOrNull() 
```

**Comment by Tapac** [comment]
@jnfeinstein , please check `MultiDatabaseTest.kt` for a sample test with multiple databases

**Comment by jnfeinstein** [comment]
> @jnfeinstein , please check `MultiDatabaseTest.kt` for a sample test with multiple databases

Done!

**Comment by Tapac** [comment]
One more thing to go, can you please add some documentation on `defaultDatabase` parameter to help users understand how to use it?

**Comment by jnfeinstein** [comment]
@Tapac no problem.  Where should this go?

**Comment by Tapac** [comment]
I guess that simple kdoc will be enough but if you want to add a few words on wiki please check [how to contribute](https://github.com/JetBrains/Exposed/wiki/Contributing#how-to-improve-wiki-pages)

**Comment by jnfeinstein** [comment]
https://github.com/Tapac/exposed-wiki/pull/13

**Comment by Tapac** [comment]
Thank you, I'll merge wiki after the Exposed release with your changes.

**Pull request merged**